apollo-ios/Sources/Apollo/ApolloStore.swift

```diff
@@ -15,13 +13,66 @@ public protocol ApolloStoreSubscriber: AnyObject {
   ///   - store: The store which made the changes
   ///   - changedKeys: The list of changed keys
   ///   - contextIdentifier: [optional] A unique identifier for the request that kicked off this change, to assist in de-duping cache hits for watchers.
+  /// @deprecated
   func store(_ store: ApolloStore,
              didChangeKeys changedKeys: Set<CacheKey>,
              contextIdentifier: UUID?)
+
+  /// A callback that can be received by subscribers for a particular activity case described in `ApolloStore.Activity`
+  ///
+  /// - Parameters:
+  ///   - store: The store which made the changes
+  ///   - activity: The activity that triggered this callback
+  ///   - contextIdentifier: [optional] A unique identifier for the request that kicked off this change, to assist in de-duping cache hits for watchers.
+  func store(_ store: ApolloStore,
+            activity: ApolloStore.Activity,
+            contextIdentifier: UUID?) throws
 }
 
 /// The `ApolloStore` class acts as a local cache for normalized GraphQL results.
 public class ApolloStore {
+  public enum Activity: Hashable {
```

Comment:
I believe that this should be a delegate with default implementations:

- The delegate pattern is well established in Apple frameworks.
- The delegate pattern, in this instance, would work well whether an app is using SwiftUI, UIKit, AppKit, or something else entirely.
- The delegate pattern would not introduce a breaking change (or require implementers to add an `@unknown default` case).